iterator_test.go

```diff
@@ -2737,3 +2738,179 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
+	var iterationsPerStep = []int{40_000, 40_000, 80_000, 80_000, 80_000, 200_000, 400_000, 1_000_000, 80_000, 80_000, 80_000}
+
+	// Should be large enough to assign a unique key to each queue.
+	const maxQueueIDLen = 1
+	const maxItemLen = 5
+	const maxKeyLen = maxQueueIDLen + 1 + maxItemLen
+	queueIDKeyspace := testkeys.Alpha(maxQueueIDLen)
+	itemKeyspace := testkeys.Alpha(maxItemLen)
+	key := make([]byte, maxKeyLen+1+len(fmt.Sprint(b.N)))
+	val := make([]byte, maxValSize)
+
+	o := (&Options{
+		DisableWAL:         true,
+		FS:                 vfs.NewMem(),
+		Comparer:           testkeys.Comparer,
+		FormatMajorVersion: FormatNewest,
+	}).EnsureDefaults()
+
+	d, err := Open("", o)
+	require.NoError(b, err)
+
+	type Queue struct {
+		deleteRatio float32
+		valueSize   int
+		start       int
+		end         int // exclusive
+	}
+
+	getKey := func(q int, i int) []byte {
+		n := testkeys.WriteKey(key, queueIDKeyspace, int64(q))
+		key[n] = '/'
+		prefixLen := n + 1
+		n = testkeys.WriteKey(key[prefixLen:], itemKeyspace, int64(i))
+		return key[:prefixLen+n]
+	}
+
+	rng := rand.New(rand.NewSource(0))
+	var queues []*Queue
+	for i := 0; i < queueCount; i++ {
+		q := &Queue{
+			// Each queue is assigned a random deleteRatio to vary the relative
+			// size of tombstone clusters that result from queue processing.
+			deleteRatio: rng.Float32() * maxDeleteRatio,
+			valueSize:   minValSize + rng.Intn(maxValSize-minValSize),
+			start:       0,
+			end:         0,
+		}
+		queues = append(queues, q)
+		b.Logf("queue=%d: deleteRatio=%f,valueSize=%d\n", i, q.deleteRatio, q.valueSize)
+	}
+
+	processQueueOnce := func(batch *Batch) {
+		for {
+			// Randomly pick a queue to process.
+			q := rng.Intn(queueCount)
+			queue := queues[q]
+
+			isDelete := rng.Float32() < queue.deleteRatio
+
+			if isDelete {
+				// Only process the queue if it's not empty. Otherwise, retry
+				// with a different queue.
+				if queue.start < queue.end {
+					err := batch.Delete(getKey(q, queue.start), nil)
+					require.NoError(b, err)
+					queue.start++
+					return
+				}
+			} else {
+				// Append to the queue.
+				err := batch.Set(getKey(q, queue.end), val, nil)
+				queue.end++
+				require.NoError(b, err)
+				return
+			}
+		}
+	}
+
+	waitForCompactions := func() {
+		d.mu.Lock()
+		d.waitTableStats()
+		for d.mu.compact.compactingCount > 0 {
+			d.mu.compact.cond.Wait()
+			d.waitTableStats()
+		}
+		d.mu.Unlock()
+	}
+
+	for step := 0; step < len(iterationsPerStep); step++ {
+		b.Run(fmt.Sprintf("step=%d", step), func(b *testing.B) {
+			var grp errgroup.Group
+
+			for proc := 0; proc < 10; proc++ {
+				grp.Go(func() error {
+					batch := d.NewBatch()
+					for i := 0; i < iterationsPerStep[step]; i++ {
+						processQueueOnce(batch)
+					}
+					if err := batch.Commit(NoSync); err != nil {
+						return err
+					}
+					return nil
+				})
+			}
+			require.NoError(b, grp.Wait())
+
+			waitForCompactions()
+
+			b.Run("queue throughput", func(b *testing.B) {
```

Comment:
Conventionally we try to not have spaces in benchmark names, so `throughput` or `queue_throughput` works here, as well as `seek` below.